LibraryCore/Enum.cs

```diff
@@ -1143,6 +1143,38 @@ public enum RefineQuality : byte
         Precise,
     }
 
+    public enum ExteriorEffect : byte
+    {
```

Comment:
If these exteriorEffect enums cross over multiple intended uses between files, can they be separated by numbers.

WhiteAura = 1

GreenWings = 20

LionWings = 40


if a new Aura was added at some point it would be better to be able to slot it in to the correct place instead of needing to add it to the end of the list.

Reply:
I don't know if i understand the 'multiple intended uses', it is only used for exterior effect... unless you refer to multiple use as the intended use is for rendering EquipEffect_Part and EquipEffect_UI.

But sure I understand the last part, I can make that change.

Reply:
Hopefully the number is adequate
```
        None,

        //Auras [1~90]
        WhiteAura = 1,
        FlameAura = 2,
        FlameAura2 = 3,
        BlueAura = 4,

        //Wings [100~200]
        GreenWings = 100,
        FlameWings = 101,
        BlueWings = 102,
        RedSinWings = 103,
        LionWings = 104,
        FireDragonWings = 105,
        SmallYellowWings = 106,
        PurpleTentacles = 107,
        GreenFeatherWings = 108,
        RedFeatherWings = 109,
        BlueFeatherWings = 110,
        WhiteFeatherWings = 111,
        BlueDragonWings = 112,
        RedWings2 = 113,
        DiamondFireWings = 114,
        PurpleTentacles2 = 115,
        PhoenixWings = 116,
        IceKingWings = 117,
        BlueButterflyWings = 118,
        AngelicWings = 119,
```

Reply:
Yes this is what I meant thank you.

Happy for emblem/weapon to also use if that's practical. 

However "weapon" effects i would suspect be mostly tied with a single weapon shape (for example the judement blade).

Reply:
Yeah I expect similar, but in hindsight the weapon effects share the same files so maybe it makes sense to keep it all together.

I realise that some more extended wing effect animations exists in EqupEffect_Full while we are only using EquipEffect_Part currently. I have modified the list again, i think this is more what you envisaged.

```
        None,

        //EquipEffect_Part [1~99] 
        WhiteAura = 1,
        FlameAura = 2,
        FlameAura2 = 3,
        BlueAura = 4,

        GreenWings = 10,
        FlameWings = 11,
        BlueWings = 12,
        RedSinWings = 13,
        LionWings = 14,
        FireDragonWings = 15,
        SmallYellowWings = 16,
        PurpleTentacles = 17,
        GreenFeatherWings = 18,
        RedFeatherWings = 19,
        BlueFeatherWings = 20,
        WhiteFeatherWings = 21,
        BlueDragonWings = 22,
        RedWings2 = 23,
        DiamondFireWings = 24,
        PurpleTentacles2 = 25,
        PhoenixWings = 26,
        IceKingWings = 27,
        BlueButterflyWings = 28,
        AngelicWings = 29,

        //EquipEffect_Full [100~119] 
        //EquipEffect_FullEx1 [120~139] 
        //EquipEffect_FullEx2 [140~159] 
        //EquipEffect_FullEx3 [160~179] 

        //MonMagicEx26 [200~250] 
        RedEyeRing = 200,
        BlueEyeRing = 201,
        GreenSpiralRing = 202,
        Fireworks = 203
```

Reply:
I wonder as well, if the weapon effect is tied to weapon shape how will you manage weapons with and without the weapon effect? It would make sense that you require duplicate weapon shapes to include with/without the effect?

Like for example 10042 = WeaponEffect, 42 = No WeaponEffect

Reply:
I think it would be nicer to keep it all in ExteriorEffect, if only we can filter the enum selection in the server console drop down based on item type....

I think it's getting a bit messy and i'm not sure how best to keep it clean for users.

Maybe for clarity we could have the convention:
Armour Effects prefixed with `A_`: A_Aura, A_Wings
Weapon Effects prefixed with 'W_`: W_HeavenBlade
Emblem Effects prefixed with `E_`: E_RedEyeRing, E_Fireworks

Reply:
Yes that makes sense prefixing them with their intended use